Gems/ROS2/Code/Source/Lidar/LidarCore.h

```diff
@@ -37,7 +37,7 @@ namespace ROS2
 
         //! Perform a raycast.
         //! @return Results of the raycast.
-        RaycastResult PerformRaycast();
+        const RaycastResult& PerformRaycast();
```

Comment:
If possible, consider
```suggestion
        const RaycastResult& PerformRaycast() const ;
```

Reply:
> If possible, consider

Probably can't make this `const` because it is returning a reference to a member function now as part of this change. The original version where a copy instance of `RaycastResult` can be `const` however.

Reply:
This method cannot be made const due to modification of the `m_lastScanResults`.